hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -463,11 +460,11 @@ private String generateUniqueCommitInstantTime(String initializationTime) {
     if (HoodieTableMetadataUtil.isIndexingCommit(dataIndexTimeline, initializationTime)) {
       return initializationTime;
     }
-    // Add suffix to initializationTime to find an unused instant time for the next index initialization.
+    // otherwise yields the timestamp on the fly.
     // This function would be called multiple times in a single application if multiple indexes are being
     // initialized one after the other.
     for (int offset = 0; ; ++offset) {
-      final String commitInstantTime = HoodieTableMetadataUtil.createIndexInitTimestamp(initializationTime, offset);
+      final String commitInstantTime = HoodieInstantTimeGenerator.instantTimePlusMillis(SOLO_COMMIT_TIMESTAMP, offset);
```

Comment:
Why do we need the SOLO_COMMIT_TIMESTAMP again here? I think the initializationTime passed to this method should already be SOLO_COMMIT_TIMESTAMP if the timeline had no completed instants right? We do this in https://github.com/apache/hudi/blob/d4f55f193cd79f317d4fc021cc741554ce8cf6cd/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java#L250

Reply:
For non-empty table, we also switch to start from the `SOLO_COMMIT_TIMESTAMP`. That's the main change.